zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/ZkClient.java

```diff
@@ -187,6 +199,10 @@ protected ZkClient(IZkConnection zkConnection, int connectionTimeout, long opera
     if (zkConnection == null) {
       throw new NullPointerException("Zookeeper connection is null!");
     }
+
+    String syncUpOnNewSession = System.getProperty(ZkSystemPropertyKeys.ZK_SYNC_UPON_NEWSESSION, "true");
```

Comment:
nit, shall we define the default value of ZK_SYNC_UPON_NEWSESSION in the ZkSystemPropertyKeys instead of hardcode it here?

Reply:
This seems not to be the normal pattern. ZkSystemPropertyKeys contains only keys, not default value.

Reply:
Or can you scratch the code or give an example?

Reply:
You are right, the keys class is just for keys. For now, we can just define a default value in this ZkClient.java (at least not hardcode).
But moving forward, we discussed that we shall have a property provider that contains the keys and default values and also reads the configuration too.

Reply:
ZK_AUTOSYNC_ENABLED_DEFAUL is define as "true"